Service/ZGWService.php

```diff
@@ -226,6 +226,36 @@ public function drcLockHandler(array $data, array $configuration): array
 
         $objectEntity = $this->entityManager->getRepository('App:ObjectEntity')->find($path['id']);
         if ($objectEntity instanceof ObjectEntity) {
+
+            // Cant unlock a non locked object.
+            if ($willBeLocked === false && $objectEntity->getValue('lock') === null) {
```

Comment:
Maybe combine all these if statements in a single function, checkForErrors or something like that.
This drcLockHandler function is to big.